pydantic_ai_slim/pydantic_ai/models/openrouter.py

```diff
@@ -0,0 +1,618 @@
+    """
+
+    openrouter_reasoning: OpenRouterReasoning
+    """To control the reasoning tokens in the request.
+
+    The reasoning config object consolidates settings for controlling reasoning strength across different models. [See more](https://openrouter.ai/docs/use-cases/reasoning-tokens)
+    """
+
+    openrouter_plugins: list[OpenRouterPlugin]
+
+
+class OpenRouterError(BaseModel):
+    """Utility class to validate error messages from OpenRouter."""
+
+    code: int
+    message: str
+
+
+class _BaseReasoningDetail(BaseModel):
+    """Common fields shared across all reasoning detail types."""
+
+    id: str | None = None
+    format: Literal['unknown', 'openai-responses-v1', 'anthropic-claude-v1', 'xai-responses-v1'] | None
+    index: int | None
+
+
+class _ReasoningSummary(_BaseReasoningDetail):
+    """Represents a high-level summary of the reasoning process."""
+
+    type: Literal['reasoning.summary']
+    summary: str = Field(validation_alias=AliasChoices('summary', 'content'))
+
+
+class _ReasoningEncrypted(_BaseReasoningDetail):
+    """Represents encrypted reasoning data."""
+
+    type: Literal['reasoning.encrypted']
+    data: str = Field(validation_alias=AliasChoices('data', 'signature'))
+
+
+class _ReasoningText(_BaseReasoningDetail):
+    """Represents raw text reasoning."""
+
+    type: Literal['reasoning.text']
+    text: str = Field(validation_alias=AliasChoices('text', 'content'))
+    signature: str | None = None
+
+
+_OpenRouterReasoningDetail = _ReasoningSummary | _ReasoningEncrypted | _ReasoningText
+_reasoning_detail_adapter: TypeAdapter[_OpenRouterReasoningDetail] = TypeAdapter(_OpenRouterReasoningDetail)
+
+
+def _from_reasoning_detail(reasoning: _OpenRouterReasoningDetail) -> ThinkingPart:
+    provider_name = 'openrouter'
+    reasoning_id = reasoning.model_dump_json(include={'id', 'format', 'index', 'type'})
+    if isinstance(reasoning, _ReasoningText):
+        return ThinkingPart(
+            id=reasoning_id,
+            content=reasoning.text,
+            signature=reasoning.signature,
+            provider_name=provider_name,
+        )
+    elif isinstance(reasoning, _ReasoningSummary):
+        return ThinkingPart(
+            id=reasoning_id,
+            content=reasoning.summary,
+            provider_name=provider_name,
+        )
+    else:
+        return ThinkingPart(
+            id=reasoning_id,
+            content='',
+            signature=reasoning.data,
+            provider_name=provider_name,
+        )
+
+
+def _into_reasoning_detail(thinking_part: ThinkingPart) -> _OpenRouterReasoningDetail:
+    if thinking_part.id is None:  # pragma: lax no cover
+        raise UnexpectedModelBehavior('OpenRouter thinking part has no ID')
+
+    data = asdict(thinking_part)
+    data.update(json.loads(thinking_part.id))
```

Comment:
If we do it like this, we should handle `ValueError`s right?

Also I have a slight preference for joining the 4 values with `|` over packing JSON into a `str` field. Did you consider that?